packages/pigeon/lib/generator_tools.dart

```diff
@@ -432,3 +432,39 @@ Iterable<EnumeratedClass> getCodecClasses(Api api, Root root) sync* {
 /// Returns true if the [TypeDeclaration] represents an enum.
 bool isEnum(Root root, TypeDeclaration type) =>
     root.enums.map((Enum e) => e.name).contains(type.baseName);
+
+/// Formats documentation comments and adds them to current Indent.
+void addDocumentationComments(Indent indent, List<String> comments, String open,
+    {String? close, List<String>? additionalComments}) {
+  if (comments.isEmpty && additionalComments == null) {
+    return;
+  }
+  if (comments.length > 1 ||
+      (comments.isNotEmpty && additionalComments != null) ||
+      (additionalComments != null && additionalComments.length > 1)) {
```

Comment:
Could you collapse this whole condition to `comments.length + additionalComments.length > 1`?